managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/InflightReadsLimiter.java

```diff
@@ -82,9 +83,10 @@ public InflightReadsLimiter(long maxReadsInFlightSize, int maxReadsInFlightAcqui
         this.remainingBytes = maxReadsInFlightSize;
         this.acquireTimeoutMillis = acquireTimeoutMillis;
         this.timeOutExecutor = timeOutExecutor;
+        this.maxReadsInFlightAcquireQueueSize = maxReadsInFlightAcquireQueueSize;
         if (maxReadsInFlightSize > 0) {
             enabled = true;
-            this.queuedHandles = new SpscArrayQueue<>(maxReadsInFlightAcquireQueueSize);
+            this.queuedHandles = new ArrayDeque<>(maxReadsInFlightAcquireQueueSize);
```

Comment:
ArrayDeque takes an initial capacity as the parameter. The purpose is `maxReadsInFlightAcquireQueueSize` is to limit the capacity. It doesn't make sense to allocate the buffer upfront. Simply remove the `maxReadsInFlightAcquireQueueSize` parameter.